value/secondary_value_order.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022-2023 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package value
+
+import (
+	"github.com/rs/zerolog/log"
+	"github.com/tigrisdata/tigris/schema"
+)
+
+/*
+https://www.mongodb.com/docs/manual/reference/bson-type-comparison-order/#std-label-bson-types-comparison-order
+MinKey (internal type)
+Null
+Numbers (ints, longs, doubles, decimals)
+Symbol, String
+Object
+Array
+BinData
+ObjectId
+Boolean - false then true
+Date
+Timestamp
+Regular Expression
+MaxKey (internal type)
+
+*/
+
+func SecondaryMinOrder() int {
+	return -1
+}
+
+func SecondaryMaxOrder() int {
+	return 127
+}
+
+func SecondaryNullOrder() int {
+	return ToSecondaryOrder(schema.NullType, nil)
+}
+
+func ToSecondaryOrder(dataType schema.FieldType, val Value) int {
+	switch dataType {
+	case schema.NullType:
+		return 5
+	case schema.DoubleType, schema.Int32Type, schema.Int64Type:
+		return 10
+	case schema.StringType, schema.UUIDType:
+		return 20
+	case schema.ObjectType:
+		return 30
+	case schema.BoolType:
+		if val != nil && val.String() == "true" {
+			return 36
+		}
+		return 35
+	case schema.DateTimeType:
+		return 40
+	case schema.MaxType:
```

Comment:
how will this be used? If type is not found can we default to MaxOrder?

Reply:
Max Type is used for querying whenever there is no end range we use this

Reply:
So they have exposed it as well?

Reply:
Yes you can do  `{field: {"lte": new MaxKey()}` or `{field: {gte: new MinKey()}}`  in js

Reply:
Interesting.